File: php/tests/ConstantTimeTest.php
<?php

declare(strict_types=1);

namespace SecureKit\Tests;

use PHPUnit\Framework\TestCase;
use SecureKit\ConstantTime;

final class ConstantTimeTest extends TestCase
{
    public function testEqualStringsMatch(): void
    {
        $this->assertTrue(ConstantTime::equals('abc123', 'abc123'));
    }

    public function testDifferentStringsDoNotMatch(): void
    {
        $this->assertFalse(ConstantTime::equals('abc123', 'abc124'));
    }

    public function testDifferentLengthStringsDoNotMatch(): void
    {
        $this->assertFalse(ConstantTime::equals('short', 'muchlongerstring'));
    }
}

File: java/src/test/java/io/github/securekit/ConstantTimeTest.java
package io.github.securekit;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class ConstantTimeTest {

    @Test
    void equalStringsMatch() {
        assertTrue(ConstantTime.equals("abc123", "abc123"));
    }

    @Test
    void differentStringsDoNotMatch() {
        assertFalse(ConstantTime.equals("abc123", "abc124"));
    }

    @Test
    void differentLengthStringsDoNotMatch() {
        assertFalse(ConstantTime.equals("short", "muchlongerstring"));
    }
}
